homeassistant/components/soma/cover.py

```diff
@@ -21,77 +38,179 @@ async def async_setup_entry(
     """Set up the Soma cover platform."""
 
     devices = hass.data[DOMAIN][DEVICES]
+    entities = []
+
+    for device in devices:
+        # Assume a shade device if the type is not present in the api response (Connect <2.2.6)
+        if ("type" in device.keys()) and (device["type"].lower() == "tilt"):
```

Comment:
```suggestion
        if "type" in device and device["type"].lower() == "tilt":
```

`keys()` is the default

Reply:
Fixed